python/pyarrow/tests/test_types.py

```diff
@@ -845,6 +845,11 @@ def test_decimal_overflow():
         with pytest.raises(ValueError):
             pa.decimal256(i, 0)
 
+def test_timedelta_overflow()::
+    d = datetime.timedelta(days=-106751992, seconds=71945, microseconds=224192)
+    with pytest.raises(pa.ArrowInvalid):
+        pa.scalar(d)
```

Comment:
The current test looks good, but will only test it for a single resolution (the default, which is micro, I think), and thus not cover all your changes. So maybe add some additional cases where you pass the data type (with specific resolution) explicitly)

Reply:
I agree it is a good idea to add test cases!

I think I am missing something, but it seems like datetime.timedelta is only at the microsecond resolution: https://docs.python.org/3/library/datetime.html#available-types

Reply:
Nvm, I think I figured it out!

Reply:
Added tests for additional resolutions